tasks/Task.cpp

```diff
@@ -301,13 +337,144 @@ base::samples::RigidBodyState Task::convertToRbs(vpHomogeneousMatrix pose)
     rbs.position[0] = pose.getTranslationVector()[0];
     rbs.position[1] = pose.getTranslationVector()[1];
     rbs.position[2] = pose.getTranslationVector()[2];
-    
+
     vpRotationMatrix R;
     pose.extract(R);
-    
     vpQuaternionVector q(R);
-    
     rbs.orientation = base::Orientation(q.w(), q.x(), q.y(), q.z());
 
     return rbs;
 }
+
+void Task::transformInput(base::LinearAngular6DCommand cmd_in_body)
+{
+    //initialize setpoint on the body frame
+    //since the non-expected input are NaN, set it to default values
+    //in order to have valid values for the HomogeneusMatrix creation
+    for (int i = 0; i < 3; ++i)
+    {
+        if (!expected_inputs.linear[i]) {cmd_in_body.linear[i] = 0;}
```

Comment:
Single line ifs should be formatted

```
if (X)
  do_y;
```

Single-line stuff is harmful in general, as it hinders code coverage measurement (there are two separate branches on the same line)